src/isar/robot/robot.py

```diff
@@ -107,10 +106,7 @@ def run(self) -> None:
         )
         self.robot_status_thread.start()
 
-        while True:
-            if self.signal_thread_quitting.is_set():
-                break
-
+        while not self.signal_thread_quitting.wait(0):
```

Comment:
This trick makes the tests more responsible when they fail (instead of hanging, they terminate faster)